### DISCUSSION TITLE
How to fix clipping data in vegetation notebook error.

### DISCUSSION THREAD

**Discussion by oye-bobs** [cu-esiil-edu/2024-data-short-course#68] [Q&A] [answered]
Hi @nquarder , I was finally able to download the data from APPEEARS today by 1:00am GMT. However on proceeding with the notebook, I ran into a problem whilst trying to  clip my DataArray to the boundaries for both inside and outside the reservation.

Here is what it looks like:
![7](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/7df9b874-4311-4750-ab20-87aca0f5d3d5)

**Comment by nquarder** [comment]
Hi @oye-bobs what is your `gdf` data look like? That might be the issue? I will try taking a look at your notebook.

**Reply by oye-bobs** [reply]
Hey, @nquarder this is what my gdf data looks like.
![8](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/b2d415fe-5567-45db-a424-95e743cecb1a)

**Comment by nquarder** [comment]
okay @oye-bobs try replacing `gdf` in the cell where you do the clipping with `cameron_peak_gdf`. You'll also need to replace `out_gdf` in the clipping cell with whatever the name of the variable you chose to call your `out_gdf` (the area outside `cameron_peak_gdf` also called the envelope). 

Try that and see what happens!

**Comment by oye-bobs** [comment]
Hi @nquarder , I did that, but it still didn't work.

**Comment by nquarder** [comment, marked as answer]
Hi @oye-bobs - I was able to look at your notebook (https://github.com/cu-esiil-edu/shortcourse-02-vegetation-oye-bobs/blob/main/vegetation.ipynb) and it appears you have some variable names that are causing the issue. 

Notice in the cell where we write the loop to extract the date from the file names and add it as a dimension. At the top of that cell we create an empty list called `ndvi_das = []`. We fill this empty list with the output from the loop. In your code you call this `ndvi_da`. You should change this to `ndvi_das` like we have in the solutions (https://cu-esiil-edu.github.io/esiil-learning-portal/shortcourse/notebooks/05-vegetation/vegetation.html).

Then, notice in the next cell down, we create a new variable called `ndvi_da` where we create a single data array from all the individual  arrays using `xr.combine`. In your code you call this variable `time-series` and then include a `combine_attr='drop'`. I would suggest changing your code from: 

`time_series = xr.combine_by_coords(ndvi_da, combine_attrs="drop")` 

to:

`ndvi_da = xr.combine_by_coords(ndvi_das, coords=['date'])`

Then you will have the correct value assigned to the `ndvi_da` variable when you go to do the plotting and the clipping. Try this and see if it helps. -NQ

**Reply by oye-bobs** [reply]
Yeah that worked @nquarder, Thank you.
